### PR TITLE
Allow manual override to PR test results

### PR DESCRIPTION
**Pull request by iarspider**
As discussed here: https://github.com/cms-sw/cmssw/issues/43143

### CODE REVIEW

**Comment by iarspider** [comment]
hold
I still need to figure out how to properly register the new labels in `add-externals-gh-labels.py`.

File: githublabels.py [hunk 1]
```diff
@@ -10,12 +10,14 @@ def get_dpg_pog(): return []
   "rejected": "e11d21",
   "info": "0000ff",
   "doc": "257fdb",
+  "manually-approved": "defb04",
```
Comment (smuzaffar) [inline githublabels.py:13]: @iarspider , I would suggest to use existing `info` so this label

File: githublabels.py [hunk 2]
```diff
@@ -10,12 +10,14 @@ def get_dpg_pog(): return []
...
 }
 
 LABEL_TYPES = {
   "pending":  LABEL_COLORS["pending"],
   "approved": LABEL_COLORS["approved"],
   "rejected": LABEL_COLORS["rejected"],
+  "manually-approved": LABEL_COLORS["manually-approved"],
```
Comment (smuzaffar) [inline githublabels.py:20]: use `info` instead of of new color

Comment (smuzaffar) [reply githublabels.py:20]: @iarspider , label will not be called `manually-approved` and we donto want it one per cmssw category, so please remove it fro here

Comment (smuzaffar) [reply githublabels.py:20]: labels should be added in `COMMON_LABELS ` dict and we should have one label for each of these `manual-override, ib-failure, external-failure` e.g `tests-manual-override`, `tests-ib-failure`. 

I would suggest to move the `TEST_IGNORE_REASON`  from process_pr.py to here and import it in process_pr

File: githublabels.py (hunk 1 above)
Comment (iarspider) [reply githublabels.py]: done

File: githublabels.py (hunk 2 above)
Comment (iarspider) [reply githublabels.py]: done

File: githublabels.py [hunk 3]
```diff
@@ -52,6 +54,9 @@ def get_dpg_pog(): return []
 for lab in TYPE_COMMANDS:
   COMMON_LABELS[lab] = TYPE_COMMANDS[lab][0]
 
+for reason in TEST_IGNORE_REASON:
+  COMMON_LABELS["tests-override-"+reason] = LABEL_COLORS["info"]
```
Comment (smuzaffar) [inline githublabels.py:58]: drop `override-` here. `test-override-manual-override` does not seems right :-)

File: process_pr.py [hunk 4]
```diff
@@ -1318,6 +1333,8 @@ def process_pr(repo_config, gh, repo, issue, dryRun, cmsbuild_user=None, force=F
     requiresTestMessage = " (tests are also fine)"
   elif "tests-rejected" in labels:
     requiresTestMessage = " (but tests are reportedly failing)"
+  elif "tests-manually-approved" in labels:
```
Comment (smuzaffar) [inline process_pr.py:1336]: `tests-manually-approved` is not a valid label

File: process_pr.py [hunk 5]
```diff
@@ -1190,6 +1201,10 @@ def process_pr(repo_config, gh, repo, issue, dryRun, cmsbuild_user=None, force=F
   # Now updated the labels.
   xlabs = ["backport", "urgent", "backport-ok", "compilation-warnings"]
   for lab in TYPE_COMMANDS: xlabs.append(lab)
+
+  if any(label.startswith("tests-override-") for label in labels):
+    labels.append("tests-approved")
```
Comment (smuzaffar) [inline process_pr.py:1206]: need to change this logic for new `test-<xyz>` labels

**Comment by iarspider** [comment]
unhold

File: process_pr.py [hunk 6]
```diff
@@ -956,6 +957,16 @@ def process_pr(repo_config, gh, repo, issue, dryRun, cmsbuild_user=None, force=F
...
+        elif REGEX_TEST_IGNORE.match(first_line) and (signatures["tests"] == "rejected"):
+          reason = REGEX_TEST_IGNORE.match(first_line)[1].strip()
+          if reason not in TEST_IGNORE_REASON:
+            reason = "-".join(reason.split())
+            if reason not in TEST_IGNORE_REASON:
+              print("Invalid ignore reason:", reason)
+              reason = ""
+
+          if reason:
+            signatures["tests"] = "override-" + reason
```
Comment (smuzaffar) [inline process_pr.py:969]: `override-` is not needed any more ... right?

**Comment by iarspider** [comment]
unhold

**Comment by iarspider** [comment]
https://github.com/cms-sw/cms-bot/pull/2106

**Pull request closed without merging**